Common/cpp/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -425,20 +420,10 @@ void NativeReanimatedModule::maybeRequestRender() {
 }
 
 void NativeReanimatedModule::onRender(double timestampMs) {
-  try {
-    std::vector<FrameCallback> callbacks = frameCallbacks;
-    frameCallbacks.clear();
-    for (auto &callback : callbacks) {
-      callback(timestampMs);
-    }
-  } catch (std::exception &e) {
-    std::string str = e.what();
-    this->errorHandler->setError(str);
-    this->errorHandler->raise();
-  } catch (...) {
-    std::string str = "OnRender error";
-    this->errorHandler->setError(str);
-    this->errorHandler->raise();
+  std::vector<FrameCallback> callbacks = frameCallbacks;
+  frameCallbacks.clear();
```

Comment:
You should be able to just move-construct `callbacks` from `frameCallbacks` here, right?
```suggestion
  std::vector<FrameCallback> callbacks = std::move(frameCallbacks);
```

Reply:
This wasn't my code so prefer not to make this change. Is this behavior defined that the original vector gets reset when moved? Typically move constructors does not specify behavior of the moved object as normally the assumption is that the moved object is never used again. Here we still want to use the vector

Reply:
It should be fine.
C++ standard says that the moved-from object is left in a "valid but unspecified state" (so it should be fine to use the old vector later), and the `std::vector` [spec](https://en.cppreference.com/w/cpp/container/vector/vector) further specifies that "After the move, `other` is guaranteed to be [empty()](https://en.cppreference.com/w/cpp/container/vector/empty)." when move-constructing a vector, so it should clear it as well.

In case you don't feel safe using a vector after moving from it, I would still suggest doing something like
```suggestion
  std::vector<FrameCallback> callbacks;
  std::swap(callbacks, frameCallbacks)
```
since this is unambiguous in what it does, but still avoids copies/allocations.

Reply:
I'd just like to avoid copying the callbacks here, since they can potentially capture a bunch of things - in this case it's a couple of `shared_ptr`s, which means unnecessary work on the atomic reference counters inside.